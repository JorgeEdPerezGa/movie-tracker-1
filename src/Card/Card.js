import React from 'react';

const Card = () => {
  return (
    <div>jdfkfjdsal</div>
  )
}

export default Card;
