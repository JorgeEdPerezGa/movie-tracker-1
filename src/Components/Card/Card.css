.card {
  margin: 5px; }

.overview {
  display: none; }

.title {
  display: none; }

.poster {
  width: 250px; }

.vote {
  display: none; }

.not-favorite {
  background-image: url("../../Imagery/like.svg");
  height: 30px;
  width: 30px;
  position: absolute;
  background-repeat: no-repeat;
  background-size: cover;
  float: right; }

.favorite {
  background-image: url("../../Imagery/like-active.svg");
  height: 30px;
  width: 30px;
  position: absolute;
  background-repeat: no-repeat;
  background-size: cover;
  float: right; }

.favorite-button {
  height: 30px;
  margin-left: -20px;
  width: 40px;
  border: none;
  background: none;
  position: absolute;
  background-repeat: no-repeat;
  background-size: cover;
  opacity: 0.8;
  float: right; }
