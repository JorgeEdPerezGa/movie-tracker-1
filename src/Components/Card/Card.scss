.card {
  margin: 5px;
}

.overview {
  display: none;
}

.title {
  display: none;
}

.poster {
  width: 250px;
}

.backdrop {
  display: none;
}

.vote {
  display: none;
}
