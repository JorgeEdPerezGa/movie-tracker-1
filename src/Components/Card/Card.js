import React from 'react';
import './Card.css';
import { browserHistory } from 'react-router-dom'
import PropTypes from 'prop-types';



const Card = (props) => {
  const favortieClass = !props.isFavorite ? 'not-favorite' : 'favorite';

  const redirect = () => {
      props.addToFavorites(props.movie)
    }

  return (
    <div className = 'card'>
      <h1 className = 'title'>{props.title}</h1>
      <p className = 'overview'>{props.overview}</p>
      <div className = 'image-container'>
        <div className = { favortieClass }>
          <button onClick={() => redirect()}
          className = 'favorite-button'></button>
        </div>
        <img className = 'poster' src={`https://image.tmdb.org/t/p/w500${props.poster}`} />
      </div>
      <p className = 'vote'>{props.vote}</p>
    </div>
  )
}

Card.propTypes = {
addToFavorites: PropTypes.func,
backdrop: PropTypes.string,
movie: PropTypes.object,
overview: PropTypes.string,
poster: PropTypes.string,
title: PropTypes.string,
user: PropTypes.array,
vote: PropTypes.numb
}

export default Card;
