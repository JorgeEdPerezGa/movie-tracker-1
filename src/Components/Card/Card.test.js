import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

let renderedCard;
beforeEach(() => {
   renderedCard = shallow(<Card />);
});

describe('Card Test', () => {

  it('should be defined', () => {
    expect(renderedCard).toBeDefined();
  });

  it('should match snap shot', () => {
    expect(renderedCard).toMatchSnapshot();
  })
})
