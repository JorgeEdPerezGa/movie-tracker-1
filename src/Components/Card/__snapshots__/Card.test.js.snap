// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test should match snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        />,
        <p
          className="overview"
        />,
        <div
          className="image-container"
        >
          <div
            className="not-favorite"
          >
            <button
              className="favorite-button"
              onClick={[Function]}
            />
          </div>
          <img
            className="poster"
            src="https://image.tmdb.org/t/p/w500undefined"
          />
        </div>,
        <p
          className="vote"
        />,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "title",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "overview",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="not-favorite"
            >
              <button
                className="favorite-button"
                onClick={[Function]}
              />
            </div>,
            <img
              className="poster"
              src="https://image.tmdb.org/t/p/w500undefined"
            />,
          ],
          "className": "image-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="favorite-button"
                onClick={[Function]}
              />,
              "className": "not-favorite",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "favorite-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "poster",
              "src": "https://image.tmdb.org/t/p/w500undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "vote",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          />,
          <p
            className="overview"
          />,
          <div
            className="image-container"
          >
            <div
              className="not-favorite"
            >
              <button
                className="favorite-button"
                onClick={[Function]}
              />
            </div>
            <img
              className="poster"
              src="https://image.tmdb.org/t/p/w500undefined"
            />
          </div>,
          <p
            className="vote"
          />,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "title",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "overview",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="not-favorite"
              >
                <button
                  className="favorite-button"
                  onClick={[Function]}
                />
              </div>,
              <img
                className="poster"
                src="https://image.tmdb.org/t/p/w500undefined"
              />,
            ],
            "className": "image-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="favorite-button"
                  onClick={[Function]}
                />,
                "className": "not-favorite",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "favorite-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "poster",
                "src": "https://image.tmdb.org/t/p/w500undefined",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "vote",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
